src/transformers/tokenization_utils_base.py

```diff
@@ -702,9 +700,20 @@ def convert_to_tensors(
             try:
                 if prepend_batch_axis:
                     value = [value]
-
                 if not is_tensor(value):
-                    tensor = as_tensor(value)
+                    if as_tensor == np.asarray:
```

Comment:
I would really refrain from touching `src/transformers/tokenization_utils_base.py` if possible. It's quite a core file, and usage are many. Especially the `is_rectangular` check, while correct I think is tricky to keep.

For the current fix I think we can find a smaller change that will still fix the warning for the pipeline.